xarray/backends/h5netcdf_.py

```diff
@@ -60,9 +61,14 @@ def _getitem(self, key):
 
 def maybe_decode_bytes(txt):
     if isinstance(txt, bytes):
-        return txt.decode("utf-8")
-    else:
-        return txt
+        try:
```

Comment:
It'd be nice to add the attribute name to the warning. Merge this function in to `_read_attributes where you have access to the attribute name? It's seems to only be used there.

Reply:
Added attribute name and originating object (can be group or variable).